internal/repo/datasource/sql_test.go

```diff
@@ -0,0 +1,321 @@
+				"account_number": 1,
+			},
+			setupFunc: func() sqlDs {
+				db, mock, err := sqlmock.New()
+				if err != nil {
+					t.Fail()
+				}
+				dB := sqlDs{
+					sqlSvc: db,
+					table:  "newTemp",
+				}
+				mock.ExpectQuery(regexp.QuoteMeta("SELECT COUNT(`transaction_id`) FROM newTemp WHERE user_id = '1234' AND account_number = 1")).WillReturnError(nil).WillReturnRows(sqlmock.NewRows([]string{"count(transaction_id)"}).AddRow("1"))
+				mock.ExpectQuery("SELECT transaction_id, account_number, user_id, amount, transfer_to, created_at, updated_at, status, type, comment FROM newTemp WHERE user_id = '1234' AND account_number = 1 ORDER BY created_at LIMIT 1 OFFSET 2 ;").WillReturnRows(sqlmock.NewRows([]string{"transaction_id", "account_number", "user_id", "amount", "transfer_to", "created_at", "updated_at", "status", "type", "comment"}).AddRow("0000-1111-2222-3333", 1, "4444-1111-2222-3333", 1000, 1234567890, time.Date(2023, time.December, 1, 1, 1, 1, 0, time.UTC), time.Date(2023, time.December, 1, 1, 1, 1, 0, time.UTC), "approved", "debit", "no comments"))
+				return dB
+			},
+			validator: func(rows []model.Transaction, count int, err error) {
+				temp := []model.Transaction{{
+					TransactionId: "0000-1111-2222-3333",
+					AccountNumber: 1,
+					UserId:        "4444-1111-2222-3333",
+					Amount:        1000,
+					TransferTo:    1234567890,
+					CreatedAt:     time.Date(2023, time.December, 1, 1, 1, 1, 0, time.UTC),
+					UpdatedAt:     time.Date(2023, time.December, 1, 1, 1, 1, 0, time.UTC),
+					Status:        "approved",
+					Type:          "debit",
+					Comment:       "no comments",
+				}}
+				if err != nil {
+					t.Errorf("Want: %v, Got: %v", nil, err)
+					return
+				}
+				if count != 1 {
+					t.Errorf("Want: %v, Got: %v", 3, count)
+					return
+				}
+				if !reflect.DeepEqual(rows, temp) {
+					t.Errorf("Want: %v, Got: %v", temp, rows)
+					return
+				}
+			},
+		},
+		{
+			name:   "FAILURE::Get:: query error",
+			filter: map[string]interface{}{"userid": "1234"},
+			setupFunc: func() sqlDs {
+				db, mock, err := sqlmock.New()
+				if err != nil {
+					t.Fail()
+				}
+				dB := sqlDs{
+					sqlSvc: db,
+					table:  "newTemp",
+				}
+				mock.ExpectQuery(regexp.QuoteMeta("SELECT COUNT(`transaction_id`) FROM newTemp WHERE userid = '1234'")).WillReturnError(nil).WillReturnRows(sqlmock.NewRows([]string{"transaction_id"}).AddRow("1").AddRow("2").AddRow("3"))
+				mock.ExpectQuery("SELECT transaction_id, account_number, user_id, amount, transfer_to, created_at, updated_at, status, type, comment FROM newTemp WHERE userid = '1234' ORDER BY created_at LIMIT 1 OFFSET 2 ;").WillReturnError(errors.New("Unknown column"))
+				return dB
+			},
+			validator: func(rows []model.Transaction, count int, err error) {
+				if !strings.Contains(err.Error(), "Unknown column") {
+					t.Errorf("Want: %v, Got: %v", "Unknown column", err)
+				}
+			},
+		},
+		{
+			name:   "FAILURE::Get:: query error::2",
```

Comment:
What is this being tested here? error::2 doesn't explain what this test case is for.

Reply:
done